onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h

```diff
@@ -96,6 +97,9 @@ struct TensorrtFuncState {
   bool int8_calibration_cache_available;
   bool dla_enable;
   int dla_core;
+  bool cuda_graph_enable;
+  cudaGraphExec_t* cuda_graph = nullptr;
```

Comment:
why is there both cuda_graph and cuda_graph_instance?
cudaGraphExec_t is already a pointer. why do we need the pointer to pointer?
i also find the naming confusing, since we use cuda_graph variable to refer to a cuda graph executable in some places and cuda graphs in other places.

Reply:
do the cudaGraphExec_t's need to be destroyed?  do we need to use unique_ptrs here?

Reply:
cuda_graph (maybe cuda_graph_ptr is a more propriate name)  is used to indicate if graph has been captured for the subgraph. If cuda graph has been there, graph capturing will be skipped in inference. So we only capture the graph once.

Reply:
yes but there is also some naming confusion between a captured graph and an executable graph (that one instantiates from a captured graph) that can be launched.
i would like to have the naming carry that distinction. so executables maybe should either have exec or instance in the name.